File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.UserDesigner;
using RepCustomXRRichText.ControlsFamily;
using System.Drawing.Design;

namespace RepCustomXRRichText {

    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            XtraReport report = new XtraReport();

            report.Bands.Add(new DetailBand());
            report.Bands[0].Controls.Add(new CustomRichText());

            report.DesignerLoaded += new DesignerLoadedEventHandler(report_DesignerLoaded);

            xrDesignPanel1.OpenReport(report);
            xrDesignPanel1.Report.DataSource = OwnerUtility.GetAllOwners();

            report.Tag = this;
        }

        void report_DesignerLoaded(object sender, DesignerLoadedEventArgs e) {
            IToolboxService tbs = (IToolboxService)xrDesignPanel1.GetService(typeof(IToolboxService));
            ToolboxItem tbi = new ToolboxItem(typeof(CustomRichText));

            tbi.DisplayName = "Custom Rich Text";
            tbs.AddToolboxItem(tbi, "Custom Controls");

            //Form form = new Form();
            //PictureBox p = new PictureBox();
            //p.Image = tbi.Bitmap;
            //form.Controls.Add(p);
            //p.Dock = DockStyle.Fill;
            //p.SizeMode = PictureBoxSizeMode.StretchImage;
            //form.ShowDialog();
        }
    }

    
    #region Owners and Pets DataSet creator
    public sealed class OwnerUtility
    {

        private OwnerUtility() { }

        public static OwnersDataSet GetAllOwners()
        {
            return new OwnersDataSet();
        }
    }
    #endregion Owners and Pets DataSet creator

    #region Owners and Pets DataSet
    public class OwnersDataSet : DataSet
    {
        public OwnersDataSet()
            : base()
        {
            DataTable owners = new DataTable("Owners");
            DataTable pets = new DataTable("Pets");

            DataSetName = "OwnersAndPets";

            owners.Columns.Add("ownerId", typeof(Int32));
            owners.Columns.Add("yearOfBirth", typeof(Int32));
            owners.Columns.Add("firstName", typeof(string));
            owners.Columns.Add("lastName", typeof(string));

            pets.Columns.Add("ownerId", typeof(Int32));
            pets.Columns.Add("petId", typeof(Int32));
            pets.Columns.Add("name", typeof(string));
            pets.Columns.Add("isNeutured", typeof(bool));
            pets.Columns.Add("type", typeof(PetType));

            Tables.AddRange(new DataTable[] { owners, pets });
            Relations.Add("OwnersPets", owners.Columns["ownerId"], pets.Columns["ownerId"]);

            owners.Rows.Add(new object[] { 1, 1950, "John", "Smith" });
            owners.Rows.Add(new object[] { 2, 1980, "Cathy", "Williams" });
            owners.Rows.Add(new object[] { 3, 1974, "Simon", "Iverson" });
            owners.Rows.Add(new object[] { 4, 1969, "Angie", "Young" });
            owners.Rows.Add(new object[] { 5, 1988, "Charles", "Adam" });
            owners.Rows.Add(new object[] { 6, 1920, "Michelle", "Jonson" });

            pets.Rows.Add(new object[] { 1, 2, "Smyth", true, PetType.Bird });
            pets.Rows.Add(new object[] { 1, 7, "Smyth2", true, PetType.Bird });
            pets.Rows.Add(new object[] { 1, 8, "Jaidan", false, PetType.Cat });

            pets.Rows.Add(new object[] { 2, 1, "Goldie", false, PetType.Fish });
            pets.Rows.Add(new object[] { 2, 3, "Willow", true, PetType.Rodent });

            pets.Rows.Add(new object[] { 3, 4, "Surge", true, PetType.Dog });

            pets.Rows.Add(new object[] { 4, 5, "Jumbo", true, PetType.Cat });
            pets.Rows.Add(new object[] { 4, 10, "Pat", true, PetType.Cat });
            pets.Rows.Add(new object[] { 4, 11, "Mike", false, PetType.Fish });
            pets.Rows.Add(new object[] { 4, 113, "Bill", false, PetType.Fish });

            pets.Rows.Add(new object[] { 5, 12, "Pete", false, PetType.Bird });
            pets.Rows.Add(new object[] { 5, 17, "Rufus", true, PetType.Dog });
        }

        #region Disable Serialization for Tables and Relations
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        public new DataTableCollection Tables
        {
            get { return base.Tables; }
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        public new DataRelationCollection Relations
        {
            get { return base.Relations; }
        }
        #endregion Disable Serialization for Tables and Relations
    }

    public enum PetType
    {
        Dog = 1,
        Cat = 2,
        Fish = 3,
        Bird = 4,
        Rodent = 5,
        Other = 6
    }
    #endregion Owners and Pets DataSet
        
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.UserDesigner
Imports RepCustomXRRichText.ControlsFamily
Imports System.Drawing.Design

Namespace RepCustomXRRichText

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim report As New XtraReport()

			report.Bands.Add(New DetailBand())
			report.Bands(0).Controls.Add(New CustomRichText())

			AddHandler report.DesignerLoaded, AddressOf report_DesignerLoaded

			xrDesignPanel1.OpenReport(report)
			xrDesignPanel1.Report.DataSource = OwnerUtility.GetAllOwners()

			report.Tag = Me
		End Sub

		Private Sub report_DesignerLoaded(ByVal sender As Object, ByVal e As DesignerLoadedEventArgs)
			Dim tbs As IToolboxService = CType(xrDesignPanel1.GetService(GetType(IToolboxService)), IToolboxService)
			Dim tbi As New ToolboxItem(GetType(CustomRichText))

			tbi.DisplayName = "Custom Rich Text"
			tbs.AddToolboxItem(tbi, "Custom Controls")

			'Form form = new Form();
			'PictureBox p = new PictureBox();
			'p.Image = tbi.Bitmap;
			'form.Controls.Add(p);
			'p.Dock = DockStyle.Fill;
			'p.SizeMode = PictureBoxSizeMode.StretchImage;
			'form.ShowDialog();
		End Sub
	End Class


	#Region "Owners and Pets DataSet creator"
	Public NotInheritable Class OwnerUtility

		Private Sub New()
		End Sub

		Public Shared Function GetAllOwners() As OwnersDataSet
			Return New OwnersDataSet()
		End Function
	End Class
	#End Region ' Owners and Pets DataSet creator

	#Region "Owners and Pets DataSet"
	Public Class OwnersDataSet
		Inherits DataSet
		Public Sub New()
			MyBase.New()
			Dim owners As New DataTable("Owners")
			Dim pets As New DataTable("Pets")

			DataSetName = "OwnersAndPets"

			owners.Columns.Add("ownerId", GetType(Int32))
			owners.Columns.Add("yearOfBirth", GetType(Int32))
			owners.Columns.Add("firstName", GetType(String))
			owners.Columns.Add("lastName", GetType(String))

			pets.Columns.Add("ownerId", GetType(Int32))
			pets.Columns.Add("petId", GetType(Int32))
			pets.Columns.Add("name", GetType(String))
			pets.Columns.Add("isNeutured", GetType(Boolean))
			pets.Columns.Add("type", GetType(PetType))

			Tables.AddRange(New DataTable() { owners, pets })
			Relations.Add("OwnersPets", owners.Columns("ownerId"), pets.Columns("ownerId"))

			owners.Rows.Add(New Object() { 1, 1950, "John", "Smith" })
			owners.Rows.Add(New Object() { 2, 1980, "Cathy", "Williams" })
			owners.Rows.Add(New Object() { 3, 1974, "Simon", "Iverson" })
			owners.Rows.Add(New Object() { 4, 1969, "Angie", "Young" })
			owners.Rows.Add(New Object() { 5, 1988, "Charles", "Adam" })
			owners.Rows.Add(New Object() { 6, 1920, "Michelle", "Jonson" })

			pets.Rows.Add(New Object() { 1, 2, "Smyth", True, PetType.Bird })
			pets.Rows.Add(New Object() { 1, 7, "Smyth2", True, PetType.Bird })
			pets.Rows.Add(New Object() { 1, 8, "Jaidan", False, PetType.Cat })

			pets.Rows.Add(New Object() { 2, 1, "Goldie", False, PetType.Fish })
			pets.Rows.Add(New Object() { 2, 3, "Willow", True, PetType.Rodent })

			pets.Rows.Add(New Object() { 3, 4, "Surge", True, PetType.Dog })

			pets.Rows.Add(New Object() { 4, 5, "Jumbo", True, PetType.Cat })
			pets.Rows.Add(New Object() { 4, 10, "Pat", True, PetType.Cat })
			pets.Rows.Add(New Object() { 4, 11, "Mike", False, PetType.Fish })
			pets.Rows.Add(New Object() { 4, 113, "Bill", False, PetType.Fish })

			pets.Rows.Add(New Object() { 5, 12, "Pete", False, PetType.Bird })
			pets.Rows.Add(New Object() { 5, 17, "Rufus", True, PetType.Dog })
		End Sub

		#Region "Disable Serialization for Tables and Relations"
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)> _
		Public Shadows ReadOnly Property Tables() As DataTableCollection
			Get
				Return MyBase.Tables
			End Get
		End Property

		<DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)> _
		Public Shadows ReadOnly Property Relations() As DataRelationCollection
			Get
				Return MyBase.Relations
			End Get
		End Property
		#End Region ' Disable Serialization for Tables and Relations
	End Class

	Public Enum PetType
		Dog = 1
		Cat = 2
		Fish = 3
		Bird = 4
		Rodent = 5
		Other = 6
	End Enum
	#End Region ' Owners and Pets DataSet

End Namespace